pkg/operator/ceph/object/user/controller.go

```diff
@@ -134,9 +211,12 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 func (r *ReconcileObjectStoreUser) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
 	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
 	reconcileResponse, cephObjectStoreUser, err := r.reconcile(request)
+	if err != nil {
+		r.updateStatus(k8sutil.ObservedGenerationNotAvailable, request.NamespacedName, k8sutil.ReconcileFailedStatus)
+		logger.Errorf("failed to reconcile %v", err)
```

Comment:
This is already handled by `ReportReconcileResult()` here: https://github.com/rook/rook/blob/df511fb58f79a55ae38813b71efe3a8eb3c464bd/pkg/operator/ceph/reporting/reporting.go#L144